src/webui/webapplication.cpp

```diff
@@ -347,6 +347,27 @@ void WebApplication::configure()
             : QLatin1String("default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; script-src 'self' 'unsafe-inline'; object-src 'none'; form-action 'self';"))
         + (m_isClickjackingProtectionEnabled ? QLatin1String(" frame-ancestors 'self';") : QLatin1String(""))
         + (m_isHttpsEnabled ? QLatin1String(" upgrade-insecure-requests;") : QLatin1String(""));
+
+    m_useCustomHeader = pref->isWebUICustomHeaderEnabled();
+    m_customHeaders.clear();
```

Comment:
I wouldn't parse the headers unless any of related options is changed.

Reply:
It is already guarded by `if (m_useCustomHeader) {}` isn't?

Reply:
Nope. Currently they're re-parsed each time the application settings are applied if "Use custom headers" is enabled even if the headers aren't changed.

Reply:
What is the point? It is not like users will update their preferences every second, will they? It will need another rarely used and duplicated content QString in memory all the time. And on each update it would still need to compare char-by-char with the value from preference class.  
Also I assume users won't need to put more than 4 custom headers in here so IMO the parsing time is negligible anyway.
Or we can use store the hash of it, but then there are still chances of collision...

Reply:
Well, it looks like you're right.